subprojects/testing-native/src/main/java/org/gradle/nativeplatform/test/xctest/internal/XCTestTestExecutionSpec.java

```diff
@@ -42,4 +45,8 @@ public File getRunScript() {
     public String getPath() {
         return path;
     }
+
+    public DefaultTestFilter getTestFilter() {
```

Comment:
What do you think about this returning a `XCTestSelection` instead and converting the TestFilter in the constructor?

Reply:
That sound pretty good.